steps/pipeline-tests-jest.yaml

```diff
@@ -0,0 +1,43 @@
+parameters:
+- name: environment
+  default: ''
+- name: domain_prefix
+  default: ''
+- name: test_file
+  default: ''
+- name: workingDirectory
+  default: ''
+- name: junit_output_dir
+  default: ''
+- name: test_title
+  default: ''
+- name: dns_zone
+  default: platform.hmcts.net
+- name: app_name
+  default: ''
+
+steps:
+- task: Bash@3
+  displayName: Run pipeline tests
+  inputs:
+    targetType: 'inline'
+    script: |
+      mkdir -p ${{ parameters.junit_output_dir }}
+      yarn
+      yarn test-junit
+    workingDirectory: '${{ parameters.workingDirectory }}'
+    failOnStderr: false
+  env:
+    JEST_JUNIT_OUTPUT_DIR: ${{ parameters.junit_output_dir }}
+    JEST_JUNIT_OUTPUT_NAME: report_$(system.jobId).xml
+    APP_NAME: ${{ parameters.app_name }}
+    DNS_ZONE: ${{ parameters.dns_zone }}
```

Comment:
The tests shouldn't have to figure this out.

the `TEST_URL` env var should be constructed by them, you could optionally pass these through if you need them for some reason though